std/assembly/dataview.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  loadUnsafeWithOffset,
+} from "./internal/arraybuffer";
+
+export class DataView {
+  readonly buffer: ArrayBuffer;
+  readonly byteOffset: i32;
+  readonly byteLength: i32;
+
+  constructor (buffer: ArrayBuffer, byteOffset: i32, byteLength: i32) {
+    this.buffer = buffer
+    this.byteOffset = byteOffset
+    this.byteLength = byteLength
+  }
+
+  @inline
+  getUint32 (byteOffset: u32, littleEndian: boolean): u32 {
+    if (littleEndian) {
+      return (
+        (loadUnsafeWithOffset<u8, u8>(this.buffer, byteOffset + 0, this.byteOffset) << 24) &
```

Comment:
Just use single call:
```ts
loadUnsafeWithOffset<u32,u32>(this.buffer, byteOffset);
```

Reply:
Won't the index then be aligned and thus read the wrong value? and aren't all three parameters requried? 🤔 

```ts
@inline export function loadUnsafeWithOffset<T,V>(buffer: ArrayBuffer, index: i32, byteOffset: i32): V {
  return <V>load<T>(changetype<usize>(buffer) + <usize>byteOffset + (<usize>index << alignof<T>()), HEADER_SIZE);
}
```

Reply:
For big endian the same. And use `bswap` built-in for reverse endian order

Reply:
> Won't the index then be aligned and thus possibly read the wrong value? 
Good catch! Need checking `<usize>buffer + byteOffset` alignment first and fallback to per-byte read with wrong alignment

Reply:
Is a per-byte access actually faster than simply doing an unaligned access? Might make sense to not use `loadUnsafeWithOffset` here but `load` directly if not.

Reply:
Should I use `bswap` when reading out individual bytes as well, or will it be faster to have two paths that reads the bytes differently?

Reply:
No. for per-byte access this not make sense. Just read in reverse order

Reply:
All loads in WASM are little endian. Probably just `if (!littleEndian) value = bswap(...)`.

Reply:
Hmm, actually, we should be able to just pass `0` for the `index` parameter, and control the offset with the `byteOffset` parameter?

```ts
  @inline
  getUint32 (byteOffset: u32, littleEndian: boolean): u32 {
    const result = loadUnsafeWithOffset<u32, u32>(this.buffer, 0, byteOffset);
    return littleEndian ? result : bswap<u32>(result);
  }

Reply:
Just use `load<i32>(changetype<usize>(this.buffer) + byteOffset, HEADER_SIZE)`. No need for `loadUnsafe` here as we are not operating on indexes but on offsets.